x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/all_exception_items_table/empty_viewer_state.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  EuiLoadingContent,
+  EuiImage,
+  EuiEmptyPrompt,
+  EuiButton,
+  useEuiTheme,
+  EuiPanel,
+} from '@elastic/eui';
+
+import { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
+import * as i18n from './translations';
+import type { ViewerState } from './reducer';
+import illustration from '../../../../common/images/illustration_product_no_results_magnifying_glass.svg';
+
+interface ExeptionItemsViewerEmptyPromptsComponentProps {
+  listType: ExceptionListTypeEnum;
+  currentState: ViewerState;
+  onCreateExceptionListItem: () => void;
+}
+
+const ExeptionItemsViewerEmptyPromptsComponent = ({
+  listType,
+  currentState,
+  onCreateExceptionListItem,
+}: ExeptionItemsViewerEmptyPromptsComponentProps): JSX.Element => {
+  const { euiTheme } = useEuiTheme();
+
+  const content = useMemo(() => {
+    switch (currentState) {
+      case 'error':
+        return (
+          <EuiEmptyPrompt
```

Comment:
Per review with @WafaaNasr  - abstract some of the code out so each switch statement returns the props to be passed into EuiEmptyPrompt.

Reply:
Going to be done as follow up - likely with Wafaa's upcoming work on the list detail page.